src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs

```diff
@@ -194,6 +196,8 @@ public static SyntaxNode DeserializeFrom(Stream stream, CancellationToken cancel
                 throw new InvalidOperationException(CodeAnalysisResources.TheStreamCannotBeReadFrom);
             }
 
+            // Report NFW to see if this is being used in the wild.
+            FatalError.ReportAndPropagate(new SerializationDeprecationException());
```

Comment:
@jasonmalinowski does this look correct wrt to reporting a good NFW we can trak on our side?

Reply:
I'd use ReportAndCatch; as discussed it doesn't change anything beyond the return code (if it were being used in an exception filter), but this as written might make people think this will actually throw this.

Reply:
not available at this layer :)